agent/hcp/telemetry/otel_sink.go

```diff
@@ -0,0 +1,189 @@
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/consul/agent/hcp/client"
+
+	gometrics "github.com/armon/go-metrics"
+	"github.com/hashicorp/go-hclog"
+	"go.opentelemetry.io/otel/attribute"
+	otelmetric "go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/instrument"
+	otelsdk "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+type OTELSinkOpts struct {
+	Reader otelsdk.Reader
+	Logger hclog.Logger
+}
+
+type OTELSink struct {
```

Comment:
This whole struct needs documenting.

Reply:
I've added some docs, and tried to be quite detailed here! let me know if there are any wording changes we could do to make it better.